packages/core/src/tools/shell.ts

```diff
@@ -220,6 +326,81 @@ export class ShellToolInvocation extends BaseToolInvocation<
         setPidCallback(pid);
       }
 
+      // Start the monitoring loop
+      const startTime = Date.now();
+      let currentTimeout = SHELL_TIMEOUT_MS;
```

Comment:
Should we only trigger the timeout if there's no new information coming from the buffer? I.e. a long build will continuously output content